phpseclib/Common/Enum.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace phpseclib\Common;
+
+abstract class Enum
+{
+    protected static $constants=null;
+
+    protected static function init()
+    {
+        if (self::$constants==null) {
+            $reflect = new \ReflectionClass(get_called_class());
```

Comment:
Me too, but I try to keep php 5.3 compatibility.  static::class - it's from 5.5.